src/libaktualizr/primary/reportqueue.cc

```diff
@@ -95,6 +98,7 @@ void ReportQueue::flushQueue() {
       }
     } else if (!response.isOk()) {
       LOG_WARNING << "Failed to post update events: " << response.getStatusStr();
+      storage->truncateReportEvents(MAX_PENDING_EVENTS);
```

Comment:
I am not sure that this is the right spot to delete/truncate the update events.

Reply:
I think we need to do it either in `ReportQueue::run()` or in `ReportQueue::enqueue` because the `ReportQueue::flushQueue` is not always invoked (if offline).

Reply:
Probably we could do it even in `SQLStorage::saveReportEvent`. E.g. something like `delete min(ID) if count() > MAX_ALLOWED`, where `MAX_ALLOWED` allowed could be just a regular define (`#define MAX_ALLOWED 10000`).

Reply:
I've simplified the overall solution, concentrating the logic in `saveReportEvent`. Looks better this way.